testdata/loader.c

```diff
@@ -166,3 +166,26 @@ __section("socket/3") int data_sections() {
 	return 0;
 }
 #endif
+
+/*
+ * Up until LLVM 14, this program results in an .rodata.cst32 section
+ * that is accessed by 'return values[i]'. For this section, no BTF is
+ * emitted. 'values' cannot be rewritten, since there is no BTF info
+ * describing the data section.
+ */
+__section("socket/4") int anon_const() {
+	volatile int ctx = 0;
+
+// 32 bytes wide results in a .rodata.cst32 section.
+#define values \
```

Comment:
Could we just fake a `.rodata.foo` via explicit section annotations? Might be more robust on newer versions of LLVM as well.

Reply:
I've tried all possible ways I could think of to force this to be emitted to a particular section, but haven't managed. Since it's not a real symbol, clang doesn't support setting the section attribute on it.